actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -376,13 +376,11 @@ def multiple_sources_tag_builder(type, sources)
         end
 
         def resolve_image_source(source, skip_pipeline)
-          if source.is_a?(Symbol) || source.is_a?(String)
-            path_to_image(source, skip_pipeline: skip_pipeline)
-          else
+          if source.respond_to? :to_model
             polymorphic_url(source)
+          else
+            path_to_image(source, skip_pipeline: skip_pipeline)
```

Comment:
Absolutely. What should it be called, though? `resolve_asset_source` won't work because "asset", as far as I understand, refers to _static_ assets. `resolve_source`?